### PR TITLE
Fix incorrect result when aggregating count BigQuery view

### LINKED ISSUE #5635: select count(*) from BigQuery views always return 0
Is this a bug or a (unimplemented) feature ? 

Steps to reproduce:

In BQ console: `create view ds.test_view as select * from ds.my_table`

In presto cli: 
`select * from bq.ds.test_view limit 10` 
returns non-empty results, 

but 
`select count(*) from bq.ds.test_view` 
always return 0.

Presto version:343
bq.properties has `bigquery.views-enabled=true`

### PR DESCRIPTION
**Pull request by ebyhr**
Fixes #5635

We can choose executing `SELECT count(*) FROM view` on BigQuery or reading number of rows from generated _pbc table. This PR is former.

### CODE REVIEW

File: presto-bigquery/src/test/java/io/prestosql/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java [hunk 1]
```diff
@@ -105,6 +105,33 @@ public void testPredicatePushdownPrunnedColumns()
...
+    {
+        BigQuery client = createBigQueryClient();
+
+        String tableName = "test.count_aggregation_table";
+        String viewName = "test.count_aggregation_view";
+
+        executeBigQuerySql(client, "DROP TABLE IF EXISTS " + tableName);
+        executeBigQuerySql(client, "DROP VIEW IF EXISTS " + viewName);
+        executeBigQuerySql(client, "CREATE TABLE " + tableName + " (a INT64, b INT64, c INT64)");
+        executeBigQuerySql(client, "INSERT INTO " + tableName + " VALUES (1, 2, 3)");
```
Comment (losipiuk) [inline presto-bigquery/src/test/java/io/prestosql/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java:119]: Can you please add two rows to a table, so the expected `count(*)` with and without filters is different?

**Review by losipiuk — APPROVED** [review]
LGTM. 
@davidrabinowitz PTAL too.

**Comment by ebyhr** [comment]
@davidrabinowitz Gentle reminder.

File: presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQuerySplitManager.java [hunk 2]
```diff
@@ -124,8 +128,22 @@ private static boolean emptyProjectionIsRequired(Optional<List<ColumnHandle>> pr
...
             }
             else {
-                // no filters, so we can take the value from the table info
-                numberOfRows = bigQueryClient.getTable(tableId).getNumRows().longValue();
+                // no filters, so we can take the value from the table info when the object is TABLE
+                TableInfo tableInfo = bigQueryClient.getTable(tableId);
+                if (tableInfo.getDefinition().getType() == TABLE) {
+                    numberOfRows = tableInfo.getNumRows().longValue();
+                }
+                else if (tableInfo.getDefinition().getType() == VIEW) {
```
Comment (davidrabinowitz) [inline presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQuerySplitManager.java:136]: Notice that unless there's some caching on the materialized tables references, then any call to count(*) will trigger a new query and create a new table - incurring additional costs on the users. Another approach is to trigger a sql `SELECT COUNT(*) FROM view` and use the result, which will give BigQuery the option to use some optimizations.

Comment (ebyhr) [reply presto-bigquery/src/main/java/io/prestosql/plugin/bigquery/BigQuerySplitManager.java:136]: As I left to PR description, I considered your "another" approach as well. If BigQuery optimizes the aggregation query internally, I would choose `COUNT(*)` query approach.

**Comment by ebyhr** [comment]
@losipiuk @davidrabinowitz Changed to the first approach (executing `count(*)`).

**Review by losipiuk — APPROVED** [review]

**Pull request merged**